From a338d37b67503712d746d08998b071bb6a73c4db Mon Sep 17 00:00:00 2001
From: You-Sheng Yang <vicamo.yang@canonical.com>
Date: Thu, 29 Aug 2019 16:46:06 +0800
Subject: [PATCH] chroot-setup: ensure gnupg is installed in chroot

In commit 693b8842e77643da2e01eaf5dc565586c2e55575 the gpg key was
installed directly into the chroot to avoid contacting keyservers, but
apt-key needs gnupg to perform the actual key import. In Bionic and
newer, apt depends on gpgv instead of gnupg and moves gnupg to Suggests,
which leaves gnupg uninstalled and fails apt update/install processes.

Signed-off-by: You-Sheng Yang <vicamo.yang@canonical.com>
---
 chroot-setup/scripts/build-mkschroot | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/chroot-setup/scripts/build-mkschroot b/chroot-setup/scripts/build-mkschroot
index 6f7ab462..c1cf54e0 100755
--- a/chroot-setup/scripts/build-mkschroot
+++ b/chroot-setup/scripts/build-mkschroot
@@ -42,6 +42,8 @@ TARGET=""
 MIRROR=""
 CL_MIRROR=""
 EMULATOR=""
+MUST_INCLUDE="gnupg"
+INCLUDE="$MUST_INCLUDE"
 
 while [ $# -ne 0 ]; do
 	case $1 in
@@ -84,6 +86,10 @@ while [ $# -ne 0 ]; do
 				;;
 			esac
 			;;
+		--include=*)
+			INCLUDE=$(echo $1|cut -d= -f2)
+			INCLUDE=${INCLUDE:+$INCLUDE,}$MUST_INCLUDE
+			;;
 		-*|--*)
 			OPTIONS="$OPTIONS $1"
 			;;
@@ -107,6 +113,8 @@ while [ $# -ne 0 ]; do
 	shift
 done
 
+OPTIONS="$OPTIONS --include=$INCLUDE"
+
 if [ "$SUITE" = "" -o "$TARGET" = "" ]; then
 	echo "$(basename $0) <suite> <target> [mirror]"
 	exit 1
-- 
2.23.0

